src/platform/cc13x2_26x2/BLEManagerImpl.cpp

```diff
@@ -1803,6 +1812,55 @@ void BLEManagerImpl::CHIPoBLEProfile_charValueChangeCB(uint8_t paramId, uint16_t
     }
 }
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+/*********************************************************************
+ * @fn      CHIPoBLEProfile_c3ValueReadCB
+ *
+ * @brief   Callback from CHIPoBLE Profile indicating a commissioner
+ *          reading an additionald commissioning-related data (C3) from
+ *          commissionee.
+ *          Calling context (BLE Stack Task)
+ *
+ * @param   destBuf - buffer to copy data to
+ *          maxLen - maximum length of destBuf
+ *
+ * @return  None.
+ */
+void BLEManagerImpl::CHIPoBLEProfile_c3ValueReadCB(uint8_t * destBuf, uint16_t maxLen, uint16_t connHandle)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    System::PacketBufferHandle packetBuf;
+    uint16_t len;
+
+    char serialNumber[ConfigurationManager::kMaxSerialNumberLength + 1];
+    uint16_t lifetimeCounter = 0;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    err = ConfigurationMgr().GetSerialNumber(serialNumber, sizeof(serialNumber));
```

Comment:
Please be aware that using serial number for generating rotating device ID is not compatible with the spec, according to the:
> It SHALL NOT contain a fixed identifier such as a serial number
...
The unique identifier SHALL consist of a randomly-generated 128-bit or longer octet string which
SHALL be programmed during factory provisioning or delivered to the device by the vendor using
secure means after a software update.
The unique identifier SHALL be protected against reading or writing over the air after initial
introduction into the device, and stay fixed during the lifetime of the device.

 I know that other platforms are also using it and I'm trying to fix this in this PR: https://github.com/project-chip/connectedhomeip/pull/15031, but it there is still some refactoring needed before it will be merged.

Reply:
@kkasperczyk-no , thanks for explaining.
Do you have an ETA on the merge? And followup, do you have a preference whether I merge this in before #15031 ?
- If merged before your PR, I can make a note that your PR must be considered before using the feature on TI platform.

(Since this feature is disabled by default (https://github.com/ti-tpan/connectedhomeip/blob/ti/ffs/examples/lock-app/cc13x2x7_26x2x7/args.gni#L39), it should at least (by default) be outside of this portion of spec).

Reply:
I believe this is ready to be merged, as all comments were addressed and I don't see any other requests from some time. CI failures were also fixed. If it comes to the preferences I would love to merge my PR, as it is already open for 2 weeks and solving conflicts took me a lot of time, but in the end both solutions works for me.

Reply:
Sounds good, I will hold off on merging my PR here, and check back in few days on #15031 .

Reply:
@ti-tpan the #15031 has been already merged. Please align this PR. Thanks for waiting!